src/Button/ButtonBase.tsx

```diff
@@ -62,40 +66,60 @@ const ButtonBase = forwardRef(
     }
 
     return (
-      <StyledButton
-        as={Component}
-        sx={sxStyles}
-        {...rest}
-        ref={innerRef}
-        data-block={block ? 'block' : null}
-        data-size={size === 'small' || size === 'large' ? size : undefined}
-        data-no-visuals={!LeadingVisual && !TrailingVisual && !TrailingAction ? true : undefined}
-      >
-        {Icon ? (
-          <Icon />
-        ) : (
-          <>
-            <Box as="span" data-component="buttonContent" sx={getAlignContentSize(alignContent)}>
-              {LeadingVisual && (
-                <Box as="span" data-component="leadingVisual" sx={{...iconWrapStyles}}>
-                  <LeadingVisual />
-                </Box>
-              )}
-              {children && <span data-component="text">{children}</span>}
-              {TrailingVisual && (
-                <Box as="span" data-component="trailingVisual" sx={{...iconWrapStyles}}>
-                  <TrailingVisual />
+      <>
+        <StyledButton
+          as={Component}
+          sx={sxStyles}
+          {...rest}
+          ref={innerRef}
+          data-block={block ? 'block' : null}
+          data-size={size === 'small' || size === 'large' ? size : undefined}
+          data-no-visuals={!LeadingVisual && !TrailingVisual && !TrailingAction ? true : undefined}
+          aria-disabled={loading ? true : undefined}
+          aria-describedby={loading ? 'loading-message' : undefined}
+        >
+          {Icon ? (
+            loading ? (
+              <Spinner size="small" />
```

Comment:
Is the spinner size always small or should it depend on the size of the button? 🤔

Reply:
Yeah, I think it looks good in the large button as well. I had the same thought 😆 I'll check again

Reply:
Bonus of keeping the same size, it matches the size of leadingVisual for this satisfying interaction: 

https://github.com/primer/react/assets/18661030/426caee6-4cb6-478f-b5f8-a84ba26e356e